testsuite/config_reader.py

```diff
@@ -1,13 +1,16 @@
 # pylint: disable=missing-docstring
 import configparser
-from flask import request
+from flask import request, session
 
 
-def get_config():
+def get_config(default='smart'):
     config = configparser.ConfigParser()
 
     try:
-        config.read_string(request.form['config'])
+        vendor = request.form.get('vendor', default).lower()
```

Comment:
Grabbing state from `request` here is kind of evil (took me a while to figure our what was happening when I was working on the websocket integratoin).